csharp/src/Ice/Ice1BinaryConnection.cs

```diff
@@ -281,7 +281,7 @@ internal Ice1BinaryConnection(ITransceiver transceiver, Endpoint endpoint, Objec
                 {
                     var request = new IncomingRequestFrame(Endpoint.Protocol,
                                                            readBuffer.Slice(Ice1Definitions.HeaderSize + 4),
-                                                           compressionStatus);
+                                                           _incomingFrameSizeMax);
```

Comment:
The idea is that compression will depend only on compress metadata for both ice1 and ice2

Reply:
It's also working fine with Ice 3.7 and most likely earlier versions of Ice: Ice does not reject a compressed reply if it did not request one. There is no checking at all.

Reply:
I realize such a scenario is not very likely since it requires to talk with a 4.0 server whose slice specifies a compressed response... but to me it would have been better to preserve the ice1 protocol semantics. I don't think it would make the code more complicated? If the 3.7 client doesn't support compression (missing bz2 library), it might now get a compressed response even though it never asked for it and it won't be able to uncompress it.